code/xcvm/lib/cow/src/msg.rs

```diff
@@ -0,0 +1,106 @@
+use alloc::string::String;
+use bounded_collections::{BoundedVec, ConstU32};
+use cosmwasm_std::{Addr, Coin, CosmosMsg, Uint64};
+use xcvm_core::{
+	AssetId, CallOrigin, DefaultXCVMPacket, Instruction, Network, Program, UserOrigin,
+};
+
+enum Decision {
+	Approve{solution_id : Option<String> },
+	/// if there was solution, users lists it lock
+	/// if no solution yet, he get all back to prevent ddos solvers
+	Cancel,
+	Reject,
+}
+
+
+pub enum ExecuteMsg {
+	SubmitIntention { intention: BoundedVec<Intention, ConstU32<4>> },	
+	/// user can cancel his intention
+	Decide { intention_id: String, decision: Decision },
+	/// user escrows his funds on interpreter on relevant chains and collects data on this chain
+	ProveFunds {},
+	// submit solutions of problems
+	SubmitSolutions { solutions: BoundedVec<(Solution, BoundedVec<Problem, ConstU32<32>>)> },
+
+	/// one may preload liquidity on various chains so that if solution uses that liqudity for limited operations set you likely get more 
+	PreloadLiquidity { },
+}
+
+pub struct Batch {}
+
+pub struct CoinAt {
+	network: Network,
+	coin: Coin,
+}
+
+
+pub struct ResultAt {
+	network: Network,
+	coin: (Amount, AssetId),
+}
+
+pub const ASSETS_LIMIT: u32 = 1;
+
+pub enum Limit {
+	// If true, then prefer to give as much as possible up to the limit of give and receive
+	// accordingly more. If false, then prefer to give as little as possible in order to receive
+	// as little as possible while receiving at least want.
+	Maximal(bool),
+	Slippage,
+}
+
+pub struct Intention {
+	/// maximum amount of assets to give from user xc accounts.
+	want: BoundedVec<CoinAt, ConstU32<ASSETS_LIMIT>>,
+	/// The minimum amount of assets which give should be exchanged for.
+	give: BoundedeVec<CoinAt, ConstU32<ASSETS_LIMIT>>,
```

Comment:
This also needs to specify (optional) amount.

Reply:
i will remove limit here, from msg interface, as we can swap many times even with single asset transport.  okey, want  is optional, right.